[FSDP] Test `named_parameters()` in forward (`use_orig_params=True`)

awgu opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#89066 [FSDP] Test `named_parameters()` in forward (`use_orig_params=True`)**

This adds a unit test following the FSDP change in https://github.com/pytorch/pytorch/pull/88781.

Commit:
[FSDP] Test `named_parameters()` in forward (`use_orig_params=True`)

[ghstack-poisoned]

fegin commented on test/distributed/fsdp/test_fsdp_use_orig_params.py:
```diff
@@ -1006,6 +1006,47 @@ def forward(self, x):
+                assert (
+                    param_shapes[0] is not None and param_shapes[1] is not None
+                ), "`param_sizes` should be set"
+                assert param_names == [
```
nit: should we use `assertEqual` and related APIs to assert? Just need to be sure that `self` has been overwritten.

fegin reviewed (approved):
LGTM, just a nit

Commit:
Update on "[FSDP] Test `named_parameters()` in forward (`use_orig_params=True`)"


This adds a unit test following the FSDP change in https://github.com/pytorch/pytorch/pull/88781.

[ghstack-poisoned]

Commit:
Update on "[FSDP] Test `named_parameters()` in forward (`use_orig_params=True`)"


This adds a unit test following the FSDP change in https://github.com/pytorch/pytorch/pull/88781.

[ghstack-poisoned]

awgu commented:
@pytorchbot merge

awgu commented:
@pytorchbot merge
